mqtt_client.c

```diff
@@ -34,23 +34,6 @@ static int mqtt_message_receive(esp_mqtt_client_handle_t client, int read_poll_t
 static void esp_mqtt_client_dispatch_transport_error(esp_mqtt_client_handle_t client);
 static esp_err_t send_disconnect_msg(esp_mqtt_client_handle_t client);
 
-static int esp_mqtt_handle_transport_read_error(int err, esp_mqtt_client_handle_t client)
```

Comment:
Suggest we keep this function to simplify the code below and for the benefits of these debug messages, as in IDF >= `v5.0` we could tell apart timeouts from EOF's

Reply:
I kept a version of `esp_mqtt_handle_transport_read_error()`.

However, I couldn't keep the original, because it's important that when called from anywhere but the first-byte read, it DOES call `esp_mqtt_client_dispatch_transport_error()` for any error including timeout/EOF, but in that first-byte read case, it should NOT call that function for timeout.

So, I did my best to make a function that would print a useful diagnostic, distinguishing timeout and connection closure and other error cases, and then call `esp_mqtt_client_dispatch_transport_error()` unconditionally. I kept the "if timeout, that's OK, never mind" logic to the first-byte read, which is the only place that needs it. I also improved the diagnostics in a few other places and verified that it looks good:

```
E (20363) mqtt_client: Read: CONNECTION_FAILED (Socket is not connected)
E (20375) mqtt_client: mqtt_process_receive: mqtt_message_receive() returned -1
D (20376) mqtt_client: Reconnect after 10000 ms
```

(Alternatively, I could have `esp_mqtt_client_dispatch_transport_error()` and `esp_mqtt_client_dispatch_transport_error_timeout_is_ok()` but the latter one would only get used in one place so that seemed silly?)